website/docs/guides/lifecycle.md

```diff
@@ -0,0 +1,172 @@
+---
+id: lifecycle
+title: Task Lifecycle
+---
+
+We have talked about how Effection operations are able to clean up after
+themselves, but how are they able to do this, and how can you implement your
+own operations which clean up after themselves?
+
+### Halt
+
+In order to understand the lifecycle of a Task, we must first understand the
+concept of halting a Task.
+
+In Effection, any task can be halted:
+
+``` javascript
+import { main } from 'effection';
+
+let task = main(function*() {
+  yield
+});
+
+task.halt();
+```
+
+Or:
+
+``` javascript
+main(function*() {
+  let task = yield spawn();
+  yield task.halt();
+});
+```
+
+Halting a Task means that the task itself is cancelled, it also causes any Task
+that has been spawned from the Task to be halted.
```

Comment:
Would it be worth mentioning that halting a task won't shut down the parent task the way an error will?

Reply:
This is a good point, and an important mechanism we haven't talked about yet, I'll try to incorporate this.